diff --git a/CMakeLists.txt b/CMakeLists.txt
index d403f31..aeb0af5 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -43,6 +43,7 @@ cmake_dependent_option(PUGIXML_BUILD_SHARED_AND_STATIC_LIBS
 option(PUGIXML_WCHAR_MODE "Enable wchar_t mode" OFF)
 option(PUGIXML_COMPACT "Enable compact mode" OFF)
 option(PUGIXML_INSTALL "Enable installation rules" ON)
+option(PUGIXML_HEADER_ONLY "Enable header only compilation" OFF)
 
 # Advanced options from pugiconfig.hpp
 option(PUGIXML_NO_XPATH "Disable XPath" OFF)
@@ -269,6 +270,14 @@ if(PUGIXML_INSTALL)
       "${PROJECT_SOURCE_DIR}/src/pugixml.hpp"
     DESTINATION
       ${CMAKE_INSTALL_INCLUDEDIR}${versioned-dir} COMPONENT ${PUGIXML_DEVELOPMENT_COMPONENT})
+
+   if (PUGIXML_HEADER_ONLY)
+      install(
+        FILES
+          "${PROJECT_SOURCE_DIR}/src/pugixml.cpp"
+        DESTINATION
+          ${CMAKE_INSTALL_INCLUDEDIR}${versioned-dir} COMPONENT ${PUGIXML_DEVELOPMENT_COMPONENT})
+   endif()
 endif()
 
 if (PUGIXML_BUILD_TESTS)
